### DISCUSSION TITLE
Need to call stopPropagation on Menu.Button onClick event

### DISCUSSION THREAD

**Discussion by shafkathullah** [tailwindlabs/headlessui#723] [General]
Need to call stopPropagation on Menu.Button onClick event, other wise the enclosing 'a' tag gets the clicked. Any suggestion make this happen?

**Comment by huseyinbuyukdere** [comment]
I experienced that too in below sandbox. But can you write here what do you want actually. Maybe we can find workaround solution. 

https://codesandbox.io/s/headlessui-button-example-0eviz?file=/src/App.js

**Reply by shafkathullah** [reply]
Another thing, Have you noticed, that only second click propagates the click event from Menu.Button to a tag, this is because first click is clearly handled by Menu.Button, but second is not.

**Reply by Gildaciolopes** [reply]
❤

**Comment by shafkathullah** [comment]
I think here `<Menu.Button onClick={handleClick2}>` the `onClick` listener simply not work

**Reply by huseyinbuyukdere** [reply]
Yes, But in your real case, why do you want to use button in anchor tag ?

**Reply by shafkathullah** [reply]
the UI design is I have a row data which is clickable and takes you to other page

**Reply by shafkathullah** [reply]
the lib is not giving access to event in  anyway I guess...

**Reply by shafkathullah** [reply]
Simple work around I am using now is to wrap `Menu.Button` with span and then call `stopPropagation` there, only downside is we have an additional DOM element now.

**Reply by shafkathullah** [reply]
Also like this by rendering as `Fragment` we can call `stopPropagation`

```
<Menu as="div" className="relative flex">
        <Menu.Button as={Fragment}>
          <button
            onClick={(e) => {
              e.stopPropagation();
            }}
            className="px-2 h-full bg-prep-100 rounded-r-sm text-white hover:bg-opacity-90"
          >
            <ChevronDownIcon className="block w-5 h-5" />
          </button>
        </Menu.Button>
...
</Menu>
```
I settle for this now.

**Comment by rajesh-dinewise** [comment]
is there any proper support for this now?

**Reply by rajesh-dinewise** [reply]
Seems to be a basic thing a library like this should support.
